### PR TITLE
CEED v5.0 release

### PR DESCRIPTION
**Pull request by cwsmith**
Version 5.0 of the CEED package collection.

TODO

- [x] Test `spack install ceed@5.0.0` on some machines
- [x] Merge https://github.com/spack/spack/pull/29531
- [x] Add the Magma dependency for `ceed@5.0.0`
- [x] Update Nek-related Spack packages with the latest versions
- [x] Add all the Nek-related dependencies for `ceed@5.0.0`
- [x] Merge https://github.com/spack/spack/pull/29474
- [x] Add the Ratel dependency for `ceed@5.0.0`
- [x] Merge https://github.com/spack/spack/pull/29658
- [x] Merge https://github.com/spack/spack/pull/29970?
- [x] Merge https://github.com/spack/spack/pull/29820
- [x] Add FMS
- [x] Update GSLIB version
- [x] Add FMS variant to MFEM

### CODE REVIEW

File: var/spack/repos/builtin/packages/ceed/package.py [hunk 1]
```diff
@@ -37,6 +38,17 @@ class Ceed(BundlePackage, CudaPackage, ROCmPackage):
...
+    # ceed 5.0
+    depends_on('libceed@0.10~cuda', when='@5.0.0~cuda')
+    for arch in CudaPackage.cuda_arch_values:
+        depends_on('libceed@0.10+cuda+magma cuda_arch={0}'.format(arch),
+                   when='@5.0.0+cuda cuda_arch={0}'.format(arch))
+    depends_on('libceed@0.10~rocm', when='@5.0.0~rocm')
+    for target in ROCmPackage.amdgpu_targets:
+        depends_on('libceed@0.10+rocm amdgpu_target={0}'.format(target),
+                   when='@5.0.0+rocm amdgpu_target={0}'.format(target))
+    depends_on('libceed@0.10+occa', when='@5.0.0+occa')
```
Comment (jeremylt) [inline var/spack/repos/builtin/packages/ceed/package.py]: The libCEED OCCA backend is experimental and unmaintaned. We should drop any use of libceed+occa in the CEED spack package

**Comment by v-dobrev** [comment]
The `fms` package name is already taken in Spack: see https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/fms/package.py which is this project: https://github.com/NOAA-GFDL/FMS.

What should we call our package? Here are some ideas:
* `ceed-fms`
* `fms-ceed`
* `fms-fem`
* `fms-format`
* ...

**Comment by tzanio** [comment]
I vote for `fms-format`

**Comment by v-dobrev** [comment]
Other options are: `fmslib`, `fms-lib`, or `libfms`.

**Comment by cwsmith** [comment]
⬆️ `libfms`

**Comment by jeremylt** [comment]
The Ratel spack package PR has been merged

**Comment by v-dobrev** [comment]
I added the Nek-related dependencies to the ceed package, however I run into issues when I try to install it on Mac with `spack install ceed%apple-clang` (this is with apple-clang 13.0.0 for most packages, and GCC 11.2 for the packages that need GCC and fortran). The packages that fail are NekCEM and NekRS.

**Comment by stomov** [comment]
@v-dobrev MAGMA v2.6.2 is now in spack (I closed https://github.com/spack/spack/pull/29531 because we had it merged already through another pull request).

**Comment by v-dobrev** [comment]
I added the MAGMA dependency for `ceed@5.0.0`, however, I have not tested it.

**Comment by v-dobrev** [comment]
I successfully ran `spack install ceed` on LC's Quartz machine with GCC v8.3.1.

**Comment by v-dobrev** [comment]
@cwsmith, should we enable `omega-h+cuda` when building `ceed+cuda`?

@jedbrown, @jeremylt, @nbeams, should we enable the MAGMA variant of `libceed` when building `ceed+rocm`, or is that not supported yet?

**Comment by cwsmith** [comment]
@v-dobrev I think we should leave it to the user to manually enable Cuda support for Omega_h given that there are compatibility issues  with cuda 11.2 and up that haven't yet been merged from upstream:
https://github.com/spack/spack/pull/29710/files#diff-0dba4a81f747fde08ec671f254060e9cb409eb7e13639e81ad4ada3d7d984251R50-R53

**Comment by jeremylt** [comment]
> @jedbrown, @jeremylt, @nbeams, should we enable the MAGMA variant of `libceed` when building `ceed+rocm`, or is that not supported yet?

libCEED itself tests the MAGMA+ROCm combo and it works fine for us. @nbeams would that `__half` issue be a problem here in the Spack install?

**Comment by nbeams** [comment]
>  @nbeams would that __half issue be a problem here in the Spack install?

Unfortunately, I think so, since the change to the `magma_types` header to use `__half` is in v2.6.2.  

(We need to decide how MAGMA wants to deal with this moving forward, since it seems AMD intends `__half` to only work for C++ at the moment, and therefore any C libraries/files including MAGMA headers will have the same problem as libCEED when building with HIP.)

Edited to add: for libCEED, we may be able to work it so we don't need to include the standard MAGMA headers at all.  But that would also be in a future (libCEED) release, since we weren't aware of the problem before the current release.

**Comment by jeremylt** [comment]
Since this requires manual futzing, we should perhaps not enable this combo for the users?

**Comment by nbeams** [comment]
Yeah, it's a shame since `__half` is not even relevant to the MAGMA backend at all, but since there's not really time to do another release of either library, I think we'll have to disable it this time around.

**Review by jedbrown — DISMISSED** [review]

**Comment by v-dobrev** [comment]
With the last commit (96ebdc4), the build on Lassen:
```console
spack install ceed+cuda cuda_arch=70
```
ran successfully with CUDA 10.1.243 and GCC 8.3.1.

**Review by v-dobrev — APPROVED** [review]

**Review by tzanio — APPROVED** [review]

**Pull request merged**